### PR TITLE
Add a spinner loading prompt after 3 seconds

### PR DESCRIPTION
**Pull request by shifucun**
image

### CODE REVIEW

File: static/css/base.scss [hunk 1]
```diff
@@ -356,6 +356,28 @@ $headings-font-family: "Google Sans", "Public Sans", $font-family-sans-serif;
...
+  align-items: center;
+}
+
+#spinner-text::after {
+  content: "Still loading ...";
+  z-index: 10;
+  margin-top: 150px;
+  margin-left: 20px;
+  font-size: 20px;
+  color: white;
```
Comment (beets) [inline static/css/base.scss:376]: is white text visible enough against the loading background? could you add a screenshot

File: static/css/base.scss [hunk 2]
```diff
@@ -356,6 +356,28 @@ $headings-font-family: "Google Sans", "Public Sans", $font-family-sans-serif;
...
+  opacity: 0; /* Initially hidden */
+  animation: fadeIn 0.5s forwards 3s; /* Animation after 3s delay */
```
Comment (beets) [inline static/css/base.scss:378]: fancy :)

File: static/css/base.scss [hunk 3]
```diff
@@ -356,6 +356,28 @@ $headings-font-family: "Google Sans", "Public Sans", $font-family-sans-serif;
...
+  display: flex;
+  justify-content: center;
+  align-items: center;
+}
+
+#spinner-text::after {
+  content: "Still loading ...";
+  z-index: 10;
+  margin-top: 150px;
+  margin-left: 20px;
```
Comment (beets) [inline static/css/base.scss:374]: is the margin necessary if it's already centered?

**Review by beets — APPROVED** [review]
this is a very cool, css only solution :)

File: static/css/base.scss (hunk 3 above)
Comment (shifucun) [reply static/css/base.scss:374]: updated screenshot, can you check?

File: static/css/base.scss (hunk 1 above)
Comment (shifucun) [reply static/css/base.scss:376]: done

File: static/css/base.scss (hunk 2 above)
Comment (shifucun) [reply static/css/base.scss:378]: from genAI :)

**Review by chejennifer — APPROVED** [review]
nice!

File: static/css/base.scss (hunk 3 above)
Comment (beets) [reply static/css/base.scss:374]: i think it's better to remove since it does look off center now..

**Comment by beets** [comment]
thanks for the screenshot - looks good!

File: static/css/base.scss (hunk 3 above)
Comment (shifucun) [reply static/css/base.scss:374]: I tried that but since there is "...", the text part seems off center. The 20px is used to offset that.

**Pull request merged**